contracts/V1/StateChannelDiamondProxy/StateChannelManagerStorage.sol

```diff
@@ -19,8 +19,21 @@ contract StateChannelManagerStorage {
     mapping(bytes32 => uint) latestFork; // [channelId] -> latestFork
     //TODO* - this can map to a hash, but it also has to store keys[] and timestamp
     mapping(bytes32 => mapping(uint => ForkDataAvailability)) postedBlockCalldata; //[channelId][forkCnt].map[transactionCnt][address] -> BlockCalldata
-    //TODO* - enough only to store hash (commitiment)
-    mapping(bytes32 => Dispute) disputes; // disputes[channelId] => Dispute #only 1 dispute per fork and at a time
+    
+    // ================== Dispute on chain storage ==================
+
+    /// @dev disputes[channelId] => array of dispute commitments
+    /// @dev hash(Dspute Struct, timestamp)
+    mapping(bytes32 => bytes32[]) disputes;
+
+    /// @dev disputesData[channelId] => array of encoded dispute data
+    mapping(bytes32 => bytes32[]) dipsutesData;
```

Comment:
I forgot to add these time parameters and the stateMachineImplementation to the on-chain state in the doc, but these are static and don't change (unless through governance) - I'll put them under config

As for ever the rest you can do either:
mapping(bytes32 => State) state 
Where State is {
StateSnapshot,
BlockCalldataCommitment,
DisputeData
}
(this is cleaner) 

OR 
map individually:
channelsId -> stateSnapshot
channelsId -> BlockCalldataCommitment
channelsId -> DisputeData

Need to define all the data structures